### DISCUSSION TITLE
SSR not working

### DISCUSSION THREAD

**Discussion by rhaegar453** [vercel/next.js#19051] [Help]
Hi Folks,
I recently got started with a project on NextJS which promised server side rendering rather than client side rendering. 
I created the following website https://shivarajbakale, i can still see that i'm getting the payload and the data is page is getting rendered on the client side. Is there a way to render the entire HTML on the server side and just get the HTML page on the client side?

![image](https://user-images.githubusercontent.com/15465199/98809766-214cf600-2444-11eb-8be3-1bc89f6625fb.png)

**Comment by jamesmosier** [comment]
The link you provided unfortunately does not work. 

---

>  Is there a way to render the entire HTML on the server side and just get the HTML page on the client side?

What do you mean by this?

---

If you want to fetch your data on the server, look at the [data fetching](https://nextjs.org/docs/basic-features/data-fetching) docs. You could fetch it at build time via `getStaticProps` or if you wanted to fetch it on each users request you could use `getServerSideProps`.

**Comment by allestaire** [comment]
I came in a long way finish the project and the project SSR wont work, this is sad. tsk tsk also framework still lack of documentation.
